src/test/java/org/jabref/logic/importer/DatabaseFileLookupTest.java

```diff
@@ -1,47 +1,146 @@
 package org.jabref.logic.importer;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 
 import org.jabref.logic.importer.fileformat.BibtexImporter;
+import org.jabref.logic.util.io.DatabaseFileLookup;
 import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
 import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.metadata.MetaData;
 import org.jabref.model.util.DummyFileUpdateMonitor;
+import org.jabref.preferences.FilePreferences;
 
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 import org.mockito.Answers;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 class DatabaseFileLookupTest {
 
+    @TempDir
+    Path tempDir;
     private BibDatabase database;
+    private String txtFileDirPath;
     private Collection<BibEntry> entries;
 
     private BibEntry entry1;
     private BibEntry entry2;
+    private Path txtFileDir;
+    private MetaData metaData;
+    private BibDatabaseContext databaseContext;
+    private FilePreferences filePreferences;
+    private DatabaseFileLookup fileLookup;
 
     @BeforeEach
     void setUp() throws Exception {
-        ParserResult result = new BibtexImporter(mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor())
+        ParserResult result = new BibtexImporter(mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS),
+                new DummyFileUpdateMonitor())
                 .importDatabase(ImportDataTest.UNLINKED_FILES_TEST_BIB);
         database = result.getDatabase();
         entries = database.getEntries();
 
-        entry1 = database.getEntryByCitationKey("entry1").get();
-        entry2 = database.getEntryByCitationKey("entry2").get();
+        tempDir = Files.createTempDirectory("testDir");
```

Comment:
Please use JUnit `@TempDir` and not Files.createTEmpDirectory

Reply:
@koppor I replaced the manual creation of a temporary directory with JUnit's @TempDir annotation for automatic handling.